src/utils/ExtensionUtilities.js

```diff
@@ -1,15 +1,100 @@
-import { Ray, Matrix4, Mesh, Vector3 } from 'three';
+import { Ray, Matrix4, Mesh, Vector3, BatchedMesh } from 'three';
 import { convertRaycastIntersect } from './GeometryRayIntersectUtilities.js';
 import { MeshBVH } from '../core/MeshBVH.js';
 
 const ray = /* @__PURE__ */ new Ray();
 const direction = /* @__PURE__ */ new Vector3();
 const tmpInverseMatrix = /* @__PURE__ */ new Matrix4();
 const origMeshRaycastFunc = Mesh.prototype.raycast;
+const origBatchedRaycastFunc = BatchedMesh.prototype.raycast;
 const _worldScale = /* @__PURE__ */ new Vector3();
+const _mesh = /* @__PURE__ */ new Mesh();
+const _batchIntersects = [];
 
 export function acceleratedRaycast( raycaster, intersects ) {
 
+	if ( this.isBatchedMesh ) {
+
+		acceleratedBatchedMeshRaycast.call( this, raycaster, intersects );
+
+	} else {
+
+		acceleratedMeshRaycast.call( this, raycaster, intersects );
+
+	}
+
+}
+
+function acceleratedBatchedMeshRaycast( raycaster, intersects ) {
+
+	if ( this.batchedBoundsTree ) {
+
+		const batchedBoundsTree = this.batchedBoundsTree;
+		const drawInfo = this._drawInfo;
+		const drawRanges = this._drawRanges;
+		const matrixWorld = this.matrixWorld;
+
+		_mesh.material = this.material;
+		_mesh.geometry = this.geometry;
+
+		const oldBoundsTree = _mesh.geometry.boundsTree;
+		const oldDrawRange = _mesh.geometry.drawRange;
+
+		for ( let i = 0, l = drawInfo.length; i < l; i ++ ) {
+
+			if ( ! drawInfo[ i ].visible || ! drawInfo[ i ].active ) {
+
+				continue;
+
+			}
+
+			const geometryId = drawInfo[ i ].geometryIndex;
+
+			_mesh.geometry.boundsTree = batchedBoundsTree[ geometryId ];
+
+			this.getMatrixAt( i, _mesh.matrixWorld ).premultiply( matrixWorld );
+
+			if ( ! _mesh.geometry.boundsTree ) {
+
+				this.getBoundingBoxAt( geometryId, _mesh.geometry.boundingBox ); // I'm curios, do we actually need this?
+				this.getBoundingSphereAt( geometryId, _mesh.geometry.boundingSphere );
+
+				const drawRange = drawRanges[ geometryId ];
+				_mesh.geometry.setDrawRange( drawRange.start, drawRange.count );
+
+			}
+
+			_mesh.raycast( raycaster, _batchIntersects );
+			// acceleratedMeshRaycast.call( mesh, raycaster, _batchIntersects ); OR
```

Comment:
It's possible we may want to perform the raycasting on the MeshBVH directly here since we can optimize the case where `Raycaster.firstHitOnly = true` by skipping meshes that can't be closer. We can look at this another time, though.

Reply:
With a BVH containing all instances it would be easier :P
Or we could write in the documentation that in case of `BatchedMesh`, `raycastFirst` returns the first hit for each mesh.